detectors/node/opentelemetry-resource-detector-docker/src/detectors/DockerDetector.ts

```diff
@@ -51,21 +53,39 @@ export class DockerCGroupV1Detector implements Detector {
 
   private async _getContainerId(): Promise<string | undefined> {
     try {
-      const rawData = await DockerCGroupV1Detector.readFileAsync(
+      let rawData = await DockerDetector.readFileAsync(
         this.DEFAULT_CGROUP_PATH,
         this.UTF8_UNICODE
       );
-      const splitData = rawData.trim().split('\n');
+      let splitData = rawData.trim().split('\n');
       for (const str of splitData) {
         if (str.length >= this.CONTAINER_ID_LENGTH) {
           return str.substring(str.length - this.CONTAINER_ID_LENGTH);
         }
       }
+
+      // If this code is reached - we then check for the V2_PATH
+      rawData = await DockerDetector.readFileAsync(
+        this.DEFAULT_CGROUP_V2_PATH,
+        this.UTF8_UNICODE
+      );
+
+      splitData = rawData.trim().split('\n');
+      for (let str of splitData) {
+        if (str.includes(this.HOSTNAME)) {
```

Comment:
Suggestion: Instead of using `if` inside the for loop which adds another indentation and branching, you can use `filter`:
```js
rawData.trim().split('\n').filter(str => str.includes(this.HOSTNAME))`
```

Reply:
makes sense.

Reply:
@blumamir I believe I have addressed all commets - please let me know if there is anything else .